orion/handlers/utils_test.go

```diff
@@ -0,0 +1,55 @@
+package handlers
+
+import (
+	"context"
+	"google.golang.org/grpc"
+	"reflect"
+	"testing"
+)
+
+type service struct {
+	interceptors []grpc.UnaryServerInterceptor
+}
+
+func NewMockService(interceptors ...grpc.UnaryServerInterceptor) *service {
+	s := new(service)
+	s.interceptors = interceptors
+	return s
+}
+
+func (s *service) GetInterceptors() []grpc.UnaryServerInterceptor {
+	return s.interceptors
+}
+
+func TestGetInterceptors(t *testing.T) {
+	// this test currently verifies the Interceptors is executed at the same sequence that you passed in
+	// gRPC's normal chainUnaryInterceptor is executing interceptor in a reverse order.
+	// but Orion's chainUnaryServer ensure the interceptor execution order is normal (ensure this)
+
+	sequenceArr := make([]int, 0) // for verifying GetInterceptors execution sequence
+	expectedArr := []int{1, 2, 3}
+	interceptors := []grpc.UnaryServerInterceptor{
+		func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+			sequenceArr = append(sequenceArr, 1)
+			return handler(ctx, req)
+		},
+		func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+			sequenceArr = append(sequenceArr, 2)
+			return handler(ctx, req)
+		},
+		func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+			sequenceArr = append(sequenceArr, 3)
+			return handler(ctx, req)
+		},
+	}
+	svc := NewMockService(interceptors...)
+
+	serverInterceptor := GetInterceptors(svc, CommonConfig{NoDefaultInterceptors: true})
```

Comment:
It seems that `GetInterceptors()` is not directly or indirectly used in Orion's code base. While it's `GetInterceptorsWithMethodMiddlewares()` being used by grpc and http handler to get the interceptors to use.

If the intent is to ensure Orion calls interceptors in the normal order, should we test `GetInterceptorsWithMethodMiddlewares()` instead? Or test the private/unexposed function `chainUnaryServer()` directly?

Reply:
good call, I changed to test GetInterceptorsWithMethodMiddlewares